### PR TITLE
Expose TaskQueue REST API endpoints for a deployment

### PR DESCRIPTION
**Pull request by menivaitsi**
Needs to be merged with https://github.com/AppScale/hawkeye/pull/83

### CODE REVIEW

File: AppController/djinn.rb [hunk 1]
```diff
@@ -3512,6 +3523,7 @@ def start_api_services()
     if my_node.is_load_balancer?
       threads << Thread.new {
         start_ejabberd()
+        configure_tq_haproxy()
```
Comment (cdonati) [inline AppController/djinn.rb]: Should we wait until we implement some sort of authentication before enabling it by default?

Comment (menivaitsi) [inline AppController/djinn.rb]: Yes, I'll add that. Thanks!

**Comment by menivaitsi** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1896/

File: AppController/lib/nginx.rb [hunk 2]
```diff
@@ -153,6 +153,16 @@ def self.write_fullproxy_app_config(app_name, http_port, https_port,
 JAVA_BLOBSTORE_REDIRECTION
     end
 
+    rest_taskqueue_redirection = <<REST_TASKQUEUE_REDIRECTION
+location ~ /taskqueue/v1beta2/projects/.* {
+    proxy_pass            http://gae_#{app_name}_taskqueue;
```
Comment (cdonati) [inline AppController/lib/nginx.rb]: Is this still needed?

File: AppController/lib/nginx.rb [hunk 3]
```diff
@@ -474,6 +484,57 @@ def self.create_uaserver_config(my_ip)
...
+  # Args:
+  #   my_ip: The private IP of the current host.
+  def self.create_taskqueue_rest_config(my_ip)
+    config = <<CONFIG
+upstream taskqueue_rest_api_endpoint {
+CONFIG
+      config += <<CONFIG
+    server #{my_ip}:#{TaskQueue::HAPROXY_PORT};
+CONFIG
+    config += <<CONFIG
```
Comment (cdonati) [inline AppController/lib/nginx.rb]: Are these two concatenations needed?

File: AppController/lib/nginx.rb (hunk 2 above)
Comment (menivaitsi) [inline AppController/lib/nginx.rb]: Yes, we'll be load balancing across all TaskQueue nodes.

Comment (cdonati) [inline AppController/lib/nginx.rb]: Why do we need an nginx entry for every app, though? I thought the endpoint would be deployment-based.

**Comment by menivaitsi** [comment]
https://ocd.appscale.com:8080/job/Daily%20Build/1916/

**Pull request merged**